### PR TITLE
Kotlin `1.5.10`

### PR DESCRIPTION
**Pull request by vRallev**
I will wait until I bump the version. I don't know if the plugin would be backwards compatible. I need to test.

### CODE REVIEW

File: gradle-plugin/generate_build_properties.gradle [hunk 1]
```diff
@@ -38,10 +38,16 @@ compileKotlin {
...
+      'dokkaHtml',
+      'publishPluginJar',
+      'runKtlintCheckOverMainSourceSet',
+      'sourcesJar',
+  ]
+
+  taskNames.each { taskName ->
+    tasks.named(taskName).configure {
+      it.dependsOn(generateBuildProperties)
+    }
```
Comment (ZacSweers) [inline gradle-plugin/generate_build_properties.gradle:51]: if you just make `compileKotlin` depend on it then you should cover all of these

File: dependencies.gradle [hunk 2]
```diff
@@ -7,15 +7,14 @@ ext {
   daggerVersion = '2.34'
   espressoVersion = '3.2.0'
   kotlinVersion = project.hasProperty('square.kotlinVersion') ?
-      project.getProperty('square.kotlinVersion') : '1.4.32'
+      project.getProperty('square.kotlinVersion') : '1.5.0'
```
Comment (ZacSweers) [inline dependencies.gradle:10]:
```suggestion
      project.getProperty('square.kotlinVersion') : '1.5.10'
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -10,15 +10,15 @@ jobs:
...
+    continue-on-error: ${{ startsWith(matrix.kotlin-version, '1.6') }}
 
     strategy:
       # Run all tasks, even if some fail. Note that they don't share an output, some tasks overlap
       # which is expected. If they need to share their outputs, then we need a proper caching
       # solution.
       fail-fast: false
       matrix:
-        kotlin-version : [ 1.4.32, 1.5.0-RC ]
+        kotlin-version : [ 1.5.0 ]
```
Comment (ZacSweers) [inline .github/workflows/ci.yml:21]:
```suggestion
        kotlin-version : [ 1.5.10 ]
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -47,7 +47,7 @@ jobs:
     strategy :
       fail-fast : false
       matrix :
-        kotlin-version : [ 1.4.32, 1.5.0-RC ]
+        kotlin-version : [ 1.5.0 ]
```
Comment (ZacSweers) [inline .github/workflows/ci.yml:50]:
```suggestion
        kotlin-version : [ 1.5.10 ]
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -162,7 +162,7 @@ jobs:
       # solution.
       fail-fast: false
       matrix:
-        kotlin-version: [1.4.32, 1.5.0-RC]
+        kotlin-version: [1.5.0]
```
Comment (ZacSweers) [inline .github/workflows/ci.yml:165]:
```suggestion
        kotlin-version: [1.5.10]
```

**Review by christiandeange — APPROVED** [review]

File: compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt [hunk 6]
```diff
@@ -276,3 +276,6 @@ internal fun FqName.safePackageString(
     val suffix = if (dotSuffix) "." else ""
     "$prefix$this$suffix"
   }
+
+internal fun String.capitalizeSupport(): String = replaceFirstChar(Char::uppercaseChar)
```
Comment (0xMatthewGroves) [inline compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt:280]: Nit: Why bother with the support moniker? Just leave it as String.capitalize or String.decapitalize.

**Review by 0xMatthewGroves — APPROVED** [review]

File: compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt (hunk 6 above)
Comment (vRallev) [reply compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt:280]: You're right. When I did this here a while ago I was afraid of the conflicting overload with the standard library. But I've seen in our codebase that it isn't an issue. Will change.

**Pull request merged**